Documentation & tweaks

notlee opened the pull request:
See commit messages for details.

JakeChampion commented on MIGRATION.md:
````diff
@@ -1,11 +1,48 @@
+-        <!-- link 1 -->
+</a>
+```
+2. The markup for the "More from the FT Group" link has changed. It has been moved inside the `nav` element alongside other links, within  it's is matrix group. Unlink other matrix title element there is an extra class `o-footer__matrix-title--link`, to indicate the title contains a link. The link itself also has an extra class `o-footer__matrix-link--more`, which applies the right arrow, etc. No ...
````
Some possible typo fixes
```suggestion
2. The markup for the "More from the FT Group" link has changed. It has been moved inside the `nav` element alongside other links, within  it's matrix group. Unlike other matrix title elements there is an extra class `o-footer__matrix-title--link`, to indicate the title contains a link. The link itself also has an extra class `o-footer__matrix-link--more`, which applies the right arrow, etc. Note in the diff below some classes such as `o-footer__more-from-ft` are deleted.
```

notlee commented:
Hi @ahmadmoawad. Thanks for spending the time to work on `o-footer`  with me yesterday 😄 In a final review I found a couple of bits that needed tidying up so thought I'd go ahead and make those tweaks. Hope that's ok! I've also added migration guides so we're ready to release. If you'd like to go through and see what I've changed, see my commit messages for details. Please review and let me know what you think.

Making a major release on a widely used Origami component is not a simple first task! Thanks for your work on this so far  🙌 

FYI, if you're interested in knowing more, we have some docs about the [component lifecycle](https://origami.ft.com/spec/v1/components/#component-lifecycle) and [how we write documentation](https://origami.ft.com/docs/principles/tone-and-language/), including migration guides.

notlee commented on MIGRATION.md:
```diff
@@ -1,11 +1,48 @@
 
 Follow [the migration guide on the Origami website](https://origami.ft.com/docs/tutorials/bower-to-npm/).
 
+_Note: If you are not able to upgrade to `v8` immediately, you may temporarily skip this upgrade and instead [upgrade to `v9.0.0-bower`](MIGRATION.md#migrating-from-v8-to-v9). `v9.0.0` introduces new features which are backported in `v9.0.0-bower` to support Bower and version 2 of the Origami Build Service. We do not recommend this as you may miss future updates, and will be required to upgrade f ...
```
```suggestion
_Note: If you are not able to upgrade to `v8` immediately, you may temporarily skip this upgrade and instead [upgrade to `v9.0.0-bower`](MIGRATION.md#migrating-from-v8-to-v9). `v9.0.0` introduces new features which are backported in `v9.0.0-bower` to support Bower and version 2 of the Origami Build Service. We do not recommend this as you may miss future updates, and will be [required to upgrade by July 2022](https://origami.ft.com/blog/2021/01/18/deprecating-bower-and-origami-via-npm/)._
```
